s2fft/utils/healpix_ffts.py

```diff
@@ -693,26 +693,30 @@ def ring_phase_shifts_hp_jax(
 # Custom healpix_fft_cuda primitive
 
 
-def _healpix_fft_cuda_abstract(f, L, nside, reality, fft_type, norm):
+def _healpix_fft_cuda_abstract(f, L, nside, reality, fft_type, norm, adjoint):
     # For the forward pass, the input is a HEALPix pixel-space array of size nside^2 *
     # 12 and the output is a FTM array of shape (number of rings , width of FTM slice)
     # which is (4 * nside - 1 , 2 * L  )
     healpix_size = (nside**2 * 12,)
     ftm_size = (4 * nside - 1, 2 * L)
     if fft_type == "forward":
-        assert f.shape == healpix_size
-        return f.update(shape=ftm_size, dtype=f.dtype)
+        batch_shape = (f.shape[0],) if f.ndim == 2 else ()
+        assert (f.shape[-1],) == healpix_size
+        return f.update(shape=batch_shape + ftm_size, dtype=f.dtype)
     elif fft_type == "backward":
-        print(f"f.shape {f.shape}")
-        assert f.shape == ftm_size
-        return f.update(shape=healpix_size, dtype=f.dtype)
+        batch_shape = (f.shape[0],) if f.ndim == 3 else ()
+        assert f.shape[-2:] == ftm_size
+        return f.update(shape=batch_shape + healpix_size, dtype=f.dtype)
     else:
         raise ValueError(f"fft_type {fft_type} not recognised.")
 
 
-def _healpix_fft_cuda_lowering(ctx, f, *, L, nside, reality, fft_type, norm):
+def _healpix_fft_cuda_lowering(ctx, f, *, L, nside, reality, fft_type, norm, adjoint):
+    assert _s2fft.COMPILED_WITH_CUDA, """
+    S2FFT was compiled without CUDA support. Cuda functions are not supported.
+    Please make sure that nvcc is in your path and $CUDA_HOME is set then reinstall s2fft using pip.
+    """
```

Comment:
```suggestion
    assert _s2fft.COMPILED_WITH_CUDA, """
    S2FFT was compiled without CUDA support. Cuda functions are not supported.
    Please make sure that nvcc is in your path and $CUDA_HOME is set then reinstall s2fft using pip.
    """
```
As [`assert` statements can be disabled when Python is run with optimizations enabled](https://docs.astral.sh/ruff/rules/assert/) I would say better to raise an exception here with the same message. Possibly creating a new custom exception type

```Python
class MissingCUDASupport(Exception):

    def __init__(self):
        super().__init__(
            "S2FFT was compiled without CUDA support. CUDA functions are not supported. "
            "Please make sure that nvcc is in your path and $CUDA_HOME is set "
            "then reinstall s2fft using pip."
        )